lib/hanami/utils/kernel.rb

```diff
@@ -404,15 +413,14 @@ def self.Integer(arg)
       #   # Missing #respond_to?
       #   input = BasicObject.new
       #   Hanami::Utils::Kernel.BigDecimal(input) # => TypeError
-      def self.BigDecimal(arg)
+      def self.BigDecimal(arg, precision: BIGDECIMAL_DEFAULT_PRECISION)
         case arg
+        when ->(a) { a.respond_to?(:to_d) && a.class == Rational } then arg.to_d(precision)
```

Comment:
@AlfonsoUceda @pascalbetz This arity problem will disappear if we apply this patch:

```diff
diff --git a/lib/hanami/utils/kernel.rb b/lib/hanami/utils/kernel.rb
index eecd412..d49e2f6 100644
--- a/lib/hanami/utils/kernel.rb
+++ b/lib/hanami/utils/kernel.rb
@@ -25,14 +25,6 @@ module Hanami
       # @see Hanami::Utils::Kernel.Integer
       NUMERIC_MATCHER = /\A([\d\/\.\+iE]+|NaN|Infinity)\z/.freeze
 
-      # Default precision for bigdecimal with rational number
-      #
-      # @since x.x.x
-      # @api private
-      #
-      # @see Hanami::Utils::Kernel.BigDecimal
-      BIGDECIMAL_DEFAULT_PRECISION = 1
-
       # Coerces the argument to be an Array.
       #
       # It's similar to Ruby's Kernel.Array, but it applies further
@@ -413,12 +405,12 @@ module Hanami
       #   # Missing #respond_to?
       #   input = BasicObject.new
       #   Hanami::Utils::Kernel.BigDecimal(input) # => TypeError
-      def self.BigDecimal(arg, precision: BIGDECIMAL_DEFAULT_PRECISION)
+      def self.BigDecimal(arg)
         case arg
-        when ->(a) { a.respond_to?(:to_d) && a.class == Rational } then arg.to_d(precision)
-        when ->(a) { a.respond_to?(:to_d) } then arg.to_d
-        when Float, Complex, Rational
+        when Numeric
           BigDecimal(arg.to_s)
+        when ->(a) { a.respond_to?(:to_d) }
+          arg.to_d
         else
           BigDecimal.new(arg)
         end
diff --git a/test/kernel_test.rb b/test/kernel_test.rb
index e19eb9d..4c60d60 100644
--- a/test/kernel_test.rb
+++ b/test/kernel_test.rb
@@ -813,15 +813,9 @@ describe Hanami::Utils::Kernel do
 
       describe 'when a rational number is given' do
         let(:input)     { Rational(0.3) }
-        let(:precision) { 1 }
 
         it 'returns an BigDecimal' do
-          if Hanami::Utils.jruby?
-            @result = Hanami::Utils::Kernel.BigDecimal(input, precision: precision)
-            @result.must_equal BigDecimal.new(input, precision)
-          else
-            @result.must_equal BigDecimal.new(input.to_s)
-          end
+          @result.must_equal BigDecimal.new(input.to_s)
         end
       end
 

```